lib/src/model/directus_data.dart

```diff
@@ -0,0 +1,28 @@
+abstract class DirectusData {
+  final Map<String, dynamic> _rawReceivedData;
+  final Map<String, dynamic> updatedProperties = {};
+
+  DirectusData(this._rawReceivedData) {
+    if (!_rawReceivedData.containsKey("id")) {
+      throw Exception("id is required");
+    }
+  }
+
+  String get id => getValue(forKey: "id");
+  Map<String, dynamic> getRawData() => _rawReceivedData;
+  bool get needsSaving => updatedProperties.isNotEmpty;
+
+  setValue(dynamic value, {required String forKey}) {
+    updatedProperties[forKey] = value;
+  }
+
+  dynamic getValue({required String forKey}) {
+    return updatedProperties[forKey] ?? _rawReceivedData[forKey];
+  }
+
+  Map<String, dynamic> toMap() {
+    return Map<String, dynamic>.of(_rawReceivedData)
+      ..addAll(updatedProperties)
+      ..remove("id");
```

Comment:
Why do you remove the "id" from the map ?

Reply:
I have removed it because I think it will be better to let Directus manage the id and as the id must not changed I thought it will be better this way.

My only issue is as we force DirectusData object to have an id property, when you create a new object you have to provide a fake id which will be replace after the creation. I like the fact that we force this object to have an id but I don't like to provide a fake one.

Reply:
Maybe we could create 2 constructors : the existing one to create `DirectusItem` from existing object, that throws if no `id` is provided and another constructor named `DirectusItem.new()` that creates an empty object that can be filled later using its functions for creating new objects.

If we do that, the `id` computed property should return an optional String and will need checks before every use.